handbook/communication/rfcs/index.md

```diff
@@ -111,6 +112,10 @@ Sometimes there is information relevant to an RFC, but that information can't be
 
 Only Sourcegraph teammates are able to see the revision history of RFCs (because edit access is required for that). This means if sensitive information is found in the document all you need to do is remove that information. Comments that accidentally contain sensitive info can be deleted.
 
+## Implementing RFCs
+
+When the team decides to implement an APPROVED RFC, a tracking issue is created in GitHub with as-needed sub-issues created to break down the work required to implement the full RFC. Teams may choose to implement only a portion of an RFC, and this enables us to capture the work that has not yet been completed for later prioritization.
```

Comment:
This is not universally true because we use RFCs for non-engineering and product things. I think this is an implementation detail of the RFC and the team implementing it, and doesn't need to be documented here.

TBH, I think the pain we are experiencing now in automation is an artifact of the rough planning process and high variance in the level of detail on the published RFCs. No other team seems to have this issue so I think it is premature to add more process to our documentation.

Reply:
> TBH, I think the pain we are experiencing now in automation is an artifact of the rough planning process and high variance in the level of detail on the published RFCs. 

What @christinaforney proposed above, then, may be something good for us to experiment with for automation work going forward, before resolving to make it team wide policy.

@mrnugget: This proposal is coming in part due to how you felt recently that there were quite a few loose ends on status of previous RFCs. You brain-dumped all of those loose ends into https://github.com/sourcegraph/sourcegraph/issues/7010 which definitely helped bring clarity into the situation.

What do you think about this proposal for automation work? Would you suggest some changes or something else entirely?

Reply:
I don't think creating a tracking issue for every RFC is necessary. I think the problem is due to this:

> we use RFCs for non-engineering and product things

Some RFCs are not like others. Some of them are easy to keep track of. Example: [RFC 30 - Zoekt Horizontal Scaling](https://docs.google.com/document/d/18w8T_KzYxQye8wg1g01QpMOX4_ERTtbOxMBRYaOEkmk/edit#heading=h.rf2d3v2oo71l), and that contains multiple steps of implementation.

I think for me the biggest problem comes from meta RFCs that are like mini-roadmaps or taskslists, except that they're never checked off. Examples:

- [RFC 20: Monitoring existing campaigns](https://docs.google.com/document/d/1UY9B_kLlwRtYj-fuv7XZS1-Mu99Czx9Ojm7sgEmoQIA/edit#heading=h.t1nv3cdlfjqp)
- [RFC 36: Campaign Creation (public milestone, splitted up into multiple technical milestones)](https://docs.google.com/document/d/1DKWy2zC6_rDZzoPS7ASUpvnVDK3Zj6cf_I9Riq9KNu0/edit#heading=h.b674i7gxjses)

What these two share is that they split up the included requirements into "required" and "stretch-goals/nice-to-haves". But when planning/implementing/updating the slides or roadmap, the RFC is synonymous with the "required" things, and the stretch goals are left behind.

As I wrote on Slack, I don't have a better solution I can propose right now. But I think that maybe these meta-RFCs shouldn't be RFCs but instead still part of something like [a8n sequence of milestones](https://docs.google.com/document/d/1TDsjrCy55UTZA_NyofVssnBotTyPP6Hvbrsp__aUCfM/edit#heading=h.go9qqwdnhiyu) and then the things we know we want to tackle will be extracted into technical milestones.

Or, as proposed, instead of RFCs they're "feature" tracking issues on GitHub with taskslists and, if required, technical RFCs can be written for a subset of the tasks.

Reply:
> As I wrote on Slack, I don't have a better solution I can propose right now. But I think that maybe these meta-RFCs shouldn't be RFCs but instead still part of something like a8n sequence of milestones and then the things we know we want to tackle will be extracted into technical milestones.

100% agree. I don't think these should have been RFCs, they were more vision documents.

Reply:
I think some of the challenges of the Search UI features could have been overcome with better sub-issues on the RFCs as well. I like that it creates a step for the implementer to think about and break down the problem into implementation actions. 

We can always qualify this section to be for development RFCs that are being implemented by engineering. If this really is just an implementation detail, it is a detail that has been missed on several occasions. I had discussed this idea with @tsenart in our 1:1, so wanted to make sure to write it down. I'm happy to pull this if it really isn't helpful in this context.

Reply:
Should we abandon this change and try this out with Automation (where it makes sense) as @tsenart suggested to start?

Reply:
That sounds good to me.

Reply:
> Should we abandon this change and try this out with Automation (where it makes sense) as @tsenart suggested to start?

:+1: